File: src/main/old/deck/Deck.java
package deck;

import java.util.LinkedList;
import java.util.List;
import java.util.Random;

public class Deck {
    private List<Card> cards;
    private final Random random;

    public Deck(){
        random = new Random();
        cards=new LinkedList<>();
        for(Suit suit : Suit.values()){
            for(Rank rank : Rank.values()){
                Card newCard = new Card(rank, suit);
                cards.add(newCard);
            }
        }
    }
    public Card pickRandomCard(){
        Card card = cards.get(random.nextInt(cards.size()));
        cards.remove(card);
        return card;
    }
    public List<Card> getCards(int numberOfCards) {
        LinkedList<Card> cards = new LinkedList<>();
        for(int i=0;i<numberOfCards;i++){
            cards.add(pickRandomCard());
        }
        return cards;
    }

    public int getDeckSize() {
        return cards.size();
    }
}

File: app/src/main/scala/deck/Deck.scala
package deck

import java.util
import java.util.Random


class Deck {
  private val random = new Random()
  var cards: Seq[Card] = Vector[Card]()

  cards = Suit.values.flatMap(suit => Rank.values.filterNot(Rank.EMPTY.==).map(rank => new Card(rank, suit))).toVector

  private def pickRandomCard: Card = {
    val index = random.nextInt(cards.size)
    val card = cards(index)
    cards = cards filterNot card.==
    card
  }

  def getCards(numberOfCards: Int): Seq[Card] = {
    ((0 until numberOfCards) map (_ => pickRandomCard)).toVector
  }

  def getDeckSize: Int = cards.size
}
